pkg/controller/cdapmaster/cdapmaster_controller.go

```diff
@@ -353,6 +394,53 @@ func logbackFromFile(t string, data interface{}) (string, error) {
 	return output.String(), nil
 }
 
+// Adds a component list object in state in progress. This will cause UpdateStatus to update the
+// ready type to state "false".
+func addUpgradeComponentNotReady(master *alpha1.CDAPMaster) {
+	os := status.ObjectStatus{}
+	os.Name = "upgrade"
+	os.Status = status.StatusInProgress
+	master.Status.ComponentMeta.ComponentList.Objects = append(master.Status.ComponentMeta.ComponentList.Objects, os)
+}
+
+func launchJob(master *alpha1.CDAPMaster, rsrclabels map[string]string) ([]reconciler.Object, error) {
+	master.Status.UpgradeJobVersion = uuid.New().String()
+	master.Status.Meta.ClearCondition(upgradeFailed, upgradeStartMessage, upgradeStartMessage)
+
+	upgradeResources, err := getPreUpgradeJobResources(master, rsrclabels)
+	if err != nil {
+		master.Status.Meta.SetCondition(upgradeFailed, upgradeFailedInitMessage, upgradeFailedInitMessage)
+		return nil, err
+	}
+	return upgradeResources, nil
+}
+
+// Gets the job resources and sets status to not ready, since any time when we wish to add the job
+// to expected, the status is not ready.
+func getPreUpgradeJobResources(master *alpha1.CDAPMaster, rsrclabels map[string]string) ([]reconciler.Object, error) {
+	upgradeResources, err := getPreUpgradeResources(
+		&upgradeJobSpec{
+			Image:        master.Spec.Image,
+			JobName:      getUpgradeJobName(master),
+			HostName:     getConfigName(master, "router"),
```

Comment:
This is a bit confusion why the hostname is the config name. In fact, it is a coincidence that the config map name is the same format as the router service name. The service name is actually set from the `setTemplateValue` method, using the `ServiceRouter` constant as defined in the `constants.go`. Suggest to extract the name formatting line from the `setTemplateValue` into a private method and unify it with the `getConfigName` method.

Reply:
good point, renamed getConfigName to getResourceName and used it in setTemplateValue